src/main/java/org/opentripplanner/datastore/https/HttpsDataSourceRepository.java

```diff
@@ -0,0 +1,60 @@
+package org.opentripplanner.datastore.https;
+
+import java.net.URI;
+import javax.annotation.Nonnull;
+import org.opentripplanner.datastore.api.CompositeDataSource;
+import org.opentripplanner.datastore.api.DataSource;
+import org.opentripplanner.datastore.api.FileType;
+import org.opentripplanner.datastore.base.DataSourceRepository;
+import org.opentripplanner.datastore.base.ZipStreamDataSourceDecorator;
+
+/**
+ * This data store accesses files in read-only mode over HTTPS.
+ */
+public class HttpsDataSourceRepository implements DataSourceRepository {
+
+  @Override
+  public String description() {
+    return "HTTPS";
+  }
+
+  @Override
+  public void open() {}
+
+  @Override
+  public DataSource findSource(@Nonnull URI uri, @Nonnull FileType type) {
+    if (skipUri(uri)) {
+      return null;
+    }
+    return createSource(uri, type);
+  }
+
+  @Override
+  public CompositeDataSource findCompositeSource(@Nonnull URI uri, @Nonnull FileType type) {
+    if (skipUri(uri)) {
+      return null;
+    }
+    return createCompositeSource(uri, type);
+  }
+
+  /* private methods */
+
+  private static boolean skipUri(URI uri) {
+    return !"https".equals(uri.getScheme());
+  }
+
+  private DataSource createSource(URI uri, FileType type) {
+    return new HttpsFileDataSource(uri, type);
+  }
+
+  private CompositeDataSource createCompositeSource(URI uri, FileType type) {
+    if (uri.getPath().endsWith(".zip")) {
```

Comment:
Should we do a `HEAD` request and check the returned mime type instead? That way we could also verify that the resource exists

Reply:
Files in our bucket return the content-type "application/octet-stream" (maybe we do it wrong, it should be possible to set the correct mime-type).
We should try both: .zip extension OR mime-type="application/zip"